### ISSUE TITLE
head-dim lower than 128

### ISSUE THREAD

**Issue by XuekuanWang** [thu-ml/SageAttention#166]
hello, I see that in the code here the head_dim is limited to 128.  What is the reason?

the code is 

if head_dim_og < 64:
        q = torch.nn.functional.pad(q, (0, 64 - head_dim_og))
        k = torch.nn.functional.pad(k, (0, 64 - head_dim_og))
        v = torch.nn.functional.pad(v, (0, 64 - head_dim_og))
elif head_dim_og > 64 and head_dim_og < 128:
        q = torch.nn.functional.pad(q, (0, 128 - head_dim_og))
        k = torch.nn.functional.pad(k, (0, 128 - head_dim_og))
        v = torch.nn.functional.pad(v, (0, 128 - head_dim_og))
elif head_dim_og > 128:
        raise ValueError(f"Unsupported head_dim: {head_dim_og}")

**Comment by dujifeng** [comment]
Same question, how to make sageAttention head_dim greater than 128? for example, head_dim = 512

**Comment by implank** [comment]
same
